File: src/structural/composite/typescript/classes/Compositor.ts
import { CompositorProtocol } from './interfaces/CompositorProtocol';
import { LeafProtocol } from './interfaces/LeafProtocol';

export class Compositor implements CompositorProtocol {
  private children: LeafProtocol[] = [];

  add(...values: LeafProtocol[]): void {
    values.forEach((value) => this.children.push(value));
  }
  get price(): number {
    return this.children.reduce((sum, leaf) => sum + leaf.price, 0);
  }
}

File: src/structural/composite/python/classes/compositor.py
from classes.interfaces.component_protocol import Component


class Compositor(Component):
    def __init__(self, name: str) -> None:
        super().__init__(name)
        self.children: list[Component] = []

    def add(self, *args: Component) -> None:
        for item in args:
            self.children.append(item)

    def get_price(self) -> int:
        return sum(item.get_price() for item in self.children)
